tools/uninstall_opte.sh

```diff
@@ -63,7 +63,14 @@ function to_stock_helios {
     echo "Note that this may entail an upgrade, depending on"
     echo "how your Helios system is configured."
     pkg set-publisher --sticky --search-first "$HELIOS_PUBLISHER"
-    pkg install --no-refresh -v "$CONSOLIDATION"
+
+    # Some packages which don't exist in stock might've been pulled in and
+    # prevent us from installing the stock consolidation package. So, we'll
+    # do a dry-run and pick out any packages we need to reject.
+    local PKGS_TO_REJECT=($(pkg install -n --no-refresh -v "$CONSOLIDATION" 2>&1 | grep -o "Package '.*' must be uninstalled or upgraded" | tr -d \' | cut -d ' ' -f 2 | sed 's/^/--reject /'))
+
+    # Now install stock consolidation but reject the packages we don't want.
+    pkg install --no-refresh -v ${PKGS_TO_REJECT[@]} "$CONSOLIDATION"
```

Comment:
@jclulow This is a lil adhoc but seems to work. Or is there a better way to do this?

(The specific packages I've needed to reject are `system/gpio` and `system/library/libispi`.)

Reply:
If you're going to depend on the (unstable, for people) output of the tool like this, I would probably more tightly control the environment for this `pkg` invocation.  At a minimum I would scrub out the locale stuff; e.g.,

```sh
LANG= LC_CTYPE= LC_NUMERIC= LC_TIME= LC_COLLATE= LC_MONETARY= \
    LC_MESSAGES= LC_ALL=C.UTF-8 \
    pkg install ...
```

Otherwise it's possible the string you're looking for will be spelt differently depending on the user environment.

I suspect it may also be preferable to just hard-code the list of packages you have there which works (`system/gpio`, `system/library/libispi`) and if this step fails, print an informative message about what one might do next otherwise (i.e., add more to the list).  That would probably be less likely to result in accidentally removing a lot of packages due to some unexpected behaviour on someone's system.

Reply:
Updated it with the constrained locale env. Also added some confirmation asking the user before just trying to uninstall.